packages/agents/sdk/src/sdkBase.ts

```diff
@@ -54,87 +48,162 @@ export class NxtpSdkBase extends NxtpSdkShared {
    * @returns providers.TransactionRequest object.
    */
   async xcall(args: NxtpSdkXCallArgs): Promise<providers.TransactionRequest> {
     const { requestContext, methodContext } = createLoggingContext(this.xcall.name);
     this.logger.info("Method start", requestContext, methodContext, { args });
 
+    const {
+      origin,
+      relayerFee: _relayerFee,
+      destination,
+      asset,
+      amount: _amount,
+      slippage,
+      receiveLocal,
+      wrapNativeOnOrigin,
+      unwrapNativeOnDestination,
+    } = args;
+    let { to } = args;
+
+    // Ensure signer is provided.
     const signerAddress = this.config.signerAddress;
     if (!signerAddress) {
       throw new SignerAddressMissing();
     }
-    const { origin, relayerFee, destination, to, asset, amount, slippage, receiveLocal } = args;
-
-    // Substitute default values as needed.
-    const callData = args.callData ?? "0x";
-    const delegate = args.delegate ?? to; // Default to using the user's signer address as the delegate.
-
-    // TODO: Calculate estimate for relayer fee and include it in the call params.
+    // Get the origin chain ID.
+    const chainId = await this.getChainId(origin);
+    // Get Connext address for origin.
+    const connextContractAddress = await this.getDeploymentAddress(origin, "connext");
 
-    // Validate XCall arguments.
-    if (asset === constants.AddressZero && amount !== "0") {
+    /// MARK - Validate arguments.
+    if (asset === constants.AddressZero && _amount !== "0") {
       // TODO: Custom error.
       throw new Error("Transacting asset specified was address zero; native assets are not supported!");
     }
-    // TODO: Should additionally make sure the asset is set to address(0) if the amount is 0.
-
-    // TODO: Use ajv validator:
-    // const validateInput = ajv.compile(XTransferSchema);
-    // const validInput = validateInput(params);
-    // if (!validInput) {
-    //   const msg = validateInput.errors?.map((err: any) => `${err.instancePath} - ${err.message}`).join(",");
-    //   throw new ParamsInvalid({
-    //     paramsError: msg,
-    //     params,
-    //   });
-    // }
 
-    const ConnextContractAddress = (await this.getConnext(origin)).address;
-
-    // Get the origin chain ID.
-    let chainId = this.config.chains[origin].chainId;
-    if (!chainId) {
-      chainId = await getChainIdFromDomain(origin, this.chainData);
+    // Sanitize arguments. Substitute default values as needed.
+    const amount = BigNumber.from(_amount);
+    let callData = args.callData ?? "0x";
+    const delegate = args.delegate ?? to; // Default to using the user's signer address as the delegate.
+    const relayerFee = BigNumber.from(_relayerFee ?? "0");
+    // TODO: Calculate estimate for relayer fee and include it in the call params.
+    // TODO: Should additionally make sure the asset is set to address(0) if the amount is 0.
+    // TODO: Use ajv validator.
```

Comment:
We might want to sync our branches because I've added ajv validation on SdkBase methods and annotations for the methods in branch `sdk-validation`.